hnn_core/network.py

```diff
@@ -37,18 +37,23 @@ def read_spikes(fname, gid_dict=None):
     spike_types = []
     for file in sorted(glob(str(fname))):
         spike_trial = np.loadtxt(file, dtype=str)
-        spike_times += [list(spike_trial[:, 0].astype(float))]
-        spike_gids += [list(spike_trial[:, 1].astype(int))]
-
-        # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
-        # spike type. If reading a legacy version, validate that a gid_dict is
-        # provided.
-        if spike_trial.shape[1] == 3:
-            spike_types += [list(spike_trial[:, 2].astype(str))]
+        if spike_trial.shape[0] > 0:
+            spike_times += [list(spike_trial[:, 0].astype(float))]
+            spike_gids += [list(spike_trial[:, 1].astype(int))]
+
+            # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
+            # spike type. If reading a legacy version, validate that a gid_dict is
+            # provided.
+            if spike_trial.shape[1] == 3:
+                spike_types += [list(spike_trial[:, 2].astype(str))]
+            else:
+                if gid_dict is None:
+                    raise ValueError("gid_dict must be provided if spike types "
+                                        "are unspecified in the file %s" % (file,))
```

Comment:
This is probably causing your flake8 error.

Reply:
Just checked the travis build, there's a couple more issues:
![image](https://user-images.githubusercontent.com/55253912/99813002-2598a380-2b15-11eb-917a-e4093562d23e.png)

@samikane these should be pretty quick to fix, you can enable flake8 in the settings of your IDE (VS code?) to catch these before pushing. Also if checks are failing you can click on them and look at the traceback to see the errors. 

Satisfying flake8 while making the code readable can be more art than science at times...